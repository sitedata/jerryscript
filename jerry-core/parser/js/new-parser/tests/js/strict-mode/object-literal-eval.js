/* This test contains strict mode parser tests. */

/* without semicolon, this would be a function call. */
"use strict" ;

({ set eval(set) {} });

({ set set(eval) {} });
