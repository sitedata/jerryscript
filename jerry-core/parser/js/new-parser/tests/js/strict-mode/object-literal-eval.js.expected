
--- Script parsing start ---

  Note: switch to strict mode

  [  1] CBC_PUSH_LITERAL idx:0->string(use strict)
  [  0] CBC_POP_BLOCK
  [  1] CBC_CREATE_OBJECT

--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 1
  Flags: [small_lit_enc,strict_mode,no_lexical_env]
  Argument range end: 1
  Register range end: 1
  Identifier range end: 1
  Const literal range end: 1
  Literal range end: 1

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---

  [  1] CBC_EXT_SET_SETTER idx:1->string(eval) idx:2->function
  [  0] CBC_POP_BLOCK
  [  1] CBC_CREATE_OBJECT

--- Function parsing start ---

  Note: legacy (non-strict) argument definition


--- Script parsing end ---

Parse error 'Non strict argument definition.' at line: 8 col: 19
